schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+        thing->setStateValue("lastChargeStatus", "Overload");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusCanceledBySupervisor:
+        thing->setStateValue("lastChargeStatus", "Canceled by supervisor (external command)");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusVentilationNotAllowed:
+        thing->setStateValue("lastChargeStatus", "Ventillation not allowed");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusUnexpectedContactorOpen:
+        thing->setStateValue("lastChargeStatus", "Unexpected contactor open");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusSimplifiedMode3NotAllowed:
+        thing->setStateValue("lastChargeStatus", "Simplified mode 3 not allowed");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusPowerSupplyInternalError:
+        thing->setStateValue("lastChargeStatus", "Power supply internal error(contactor not able to close)");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusUnexpectedPlugUnlock:
+        thing->setStateValue("lastChargeStatus", "Unexpected plug unlock");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDefaultNbPhases:
+        thing->setStateValue("lastChargeStatus", "Default Nb Phases (triphase not allowed)");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDiDefaultSurgeArrestor:
+        thing->setStateValue("lastChargeStatus", "DI default Surge arrestor");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDiDefaultAntiIntrusion:
+        thing->setStateValue("lastChargeStatus", "DI default Anti Intrusion");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDiDefaultShutterUnlock:
+        thing->setStateValue("lastChargeStatus", "DI default Shutter Unlock");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDiDefaultFlsi:
+        thing->setStateValue("lastChargeStatus", "DI default FLSI (Force Load Shedding Input)");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDiDefaultEmergencyStop:
+        thing->setStateValue("lastChargeStatus", "DI default Emergency Stop");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDiDefaultUndervoltage:
+        thing->setStateValue("lastChargeStatus", "DI default Undervoltage");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusDiDefaultCi:
+        thing->setStateValue("lastChargeStatus", "DI default CI (Conditional Input)");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusOther:
+        thing->setStateValue("lastChargeStatus", "Other");
+        break;
+    case SchneiderWallboxModbusTcpConnection::LastChargeStatusUndefined:
+    default:
+        thing->setStateValue("lastChargeStatus", "Undefined");
```

Comment:
Oh, another hint. While setting states by name is supported and a perfectly valid thing to do, I would still recommend to prefer using the `setStateValue(const StateTypeId &, ...)` instead if especially when you just have a single thing class.

The reason is that setting states by name is not compile time save. Rename the lastChargeStatus state in the .json and the compiler will *not* complain this way. It would complain if you'd used the stateTypeId instead, thus preventing potential bugs with typos or renames.